pennylane/circuit_graph.py

```diff
@@ -138,16 +97,28 @@ class CircuitGraph:
     between arbitrary quantum channels and measurements, not just unitary gates.
 
     Args:
-        ops (Iterable[Operator]): quantum operators constituting the circuit, in temporal order
-        variable_deps (dict[int, list[ParameterDependency]]): Free parameters of the quantum circuit.
-            The dictionary key is the parameter index.
-        wires (Wires): the addressable wires on the device
+        ops (Iterable[.Operator]): quantum operators constituting the circuit, in temporal order
+        obs (Iterable[.MeasurementProcess]): terminal measurements, in temporal order
+        wires (.Wires): The addressable wire register of the device that will be executing this graph
+        par_info (dict[int, dict[str, .Operation or int]]): Parameter information. Keys are
+            parameter indices (in the order they appear on the tape), and values are a
+            dictionary containing the corresponding operation and operation parameter index.
+        trainable_params (set[int]): A set containing the indices of parameters that support
+            differentiability. The indices provided match the order of appearence in the
+            quantum circuit.
```

Comment:
> The indices provided match the order of appearence in the quantum circuit.

How is that ensured? It seems like sets follow an ascending order 🤔 
```python
In [4]: set([3,2,1])
Out[4]: {1, 2, 3}
```

Reply:
Oh, it is the indices _themselves_ that match the order of appearance, not their order in the provided set.

For example:

```python
with QuantumTape() as tape:
    qml.RX(0.5, wires=0)  # parameter 0
    qml.Rot(0.2, 0.1, 0.5, wires=0)  # parameters 1, 2, 3
    qml.CNOT(wires=[0, 1])
    qml.RY(0.5, wires=1) # parameter 4
```